source/aerobulk/flux.py

```diff
@@ -45,29 +207,69 @@ def flux_noskin(
         by default 2
     niter : int, optional
         Number of iteration steps used in the algorithm,
-        by default 1
+        by default 6
 
     Returns
     -------
-    ql : array_like
+    ql : xarray.DataArray
         Latent heat flux                                     [W/m^2]
-    qh : array_like
+    qh : xarray.DataArray
         Sensible heat flux                                   [W/m^2]
-    taux : array_like
+    taux : xarray.DataArray
         zonal wind stress                                    [N/m^2]
-    tauy : array_like
+    tauy : xarray.DataArray
         meridional wind stress                                    [N/m^2]
-    evap : array_like
+    evap : xarray.DataArray
         evaporation         [mm/s] aka [kg/m^2/s] (usually <0, as ocean loses water!)
     """
     _check_algo(algo, VALID_ALGOS)
-    ql, qh, taux, tauy, evap = aero.mod_aerobulk_wrapper.aerobulk_model_noskin(
-        algo, zt, zu, sst, t_zt, hum_zt, u_zu, v_zu, slp, niter
+
+    sst, t_zt, hum_zt, u_zu, v_zu, slp = xr.broadcast(
+        sst, t_zt, hum_zt, u_zu, v_zu, slp
     )
-    return ql, qh, taux, tauy, evap
 
+    if len(sst.dims) < 3:
+        # TODO promote using expand_dims?
+        raise NotImplementedError
```

Comment:
Add an actual message to this error

Reply:
I refactored these warnings into a `check_input` decorator, which we can later use to implement logic to expand/reshape the input and revert that action after the values are returned.